Break apart the frontend and backend

jlee-cl opened the pull request:
Testing is still WIP, I think this can be imported from a remote package via a git url but not exactly sure.

jlee-cl committed:
Break apart the frontend and backend

jackley-carnegielearning commented on datastore/package.json:
```diff
@@ -0,0 +1,3 @@
+{
+  "name": "@uniroster-server/datastore",
```
good.

jackley-carnegielearning commented on frontend/package.json:
```diff
@@ -0,0 +1,3 @@
+{
+  "name": "@uniroster-server/frontend"
```
good.

jackley-carnegielearning commented on frontend/routes/auth.js:
```diff
@@ -1,7 +1,7 @@
 const config = require('../config');
-const db = require('../utils/database');
+const db = require('../../datastore/utils/database');
```
is this for the oauthServer?
perhaps the database dependency for that should be split and live in frontend?

jackley-carnegielearning commented on frontend/routes/auth.js:
```diff
@@ -1,7 +1,7 @@
+const db = require('../../datastore/utils/database');
 const oauthServer = require('../oauth/server');
-const utils = require('../utils/utils');
+const utils = require('../../datastore/utils/utils');
```
same as db here?

jackley-carnegielearning commented on frontend/routes/oneroster/v1_0/academicsessions.js:
```diff
@@ -0,0 +1,29 @@
+const router = require('express').Router();
+
+const academicSessionsStore = require("@uniroster-server/datastore/oneroster/academicsessions");
```
good.

jackley-carnegielearning commented on frontend/routes/oneroster/v1_0/users.js:
```diff
@@ -1,6 +1,6 @@
-const db = require('../../../utils/database');
+const db = require('../../../../datastore/utils/database');
```
this file still mixes router and db calls. interim?

jackley-carnegielearning commented on package.json:
```diff
@@ -9,6 +9,8 @@
         "init-db:dev": "docker compose exec mariadb /uniroster-server/sql/dev_setup.sh"
     },
     "dependencies": {
+        "@uniroster-server/datastore": "file:datastore",
```
good.

jackley-carnegielearning reviewed (commented):
this looks really good.
I have questions (in comments) about the up-and-over references to the db and utils modules .

jlee-cl committed:
Fix imports

Frontend is not directly importing ../datastore but going by the package
name. Nothing is importing "../../" or deeper.
